src/lib.rs

```diff
@@ -115,34 +115,45 @@ unsafe fn console_on_any(fds: &[DWORD]) -> bool {
 
 /// Returns true if there is an MSYS tty on the given handle.
 #[cfg(windows)]
-unsafe fn msys_tty_on(fd: DWORD) -> bool {
-    use std::{mem, slice};
-
+fn msys_tty_on(fd: DWORD) -> bool {
     use winapi::{
         ctypes::c_void,
         shared::minwindef::MAX_PATH,
         um::{
-            fileapi::FILE_NAME_INFO, minwinbase::FileNameInfo, processenv::GetStdHandle,
+            minwinbase::FileNameInfo, processenv::GetStdHandle,
             winbase::GetFileInformationByHandleEx,
         },
     };
 
-    let size = mem::size_of::<FILE_NAME_INFO>();
-    let mut name_info_bytes = vec![0u8; size + MAX_PATH * mem::size_of::<WCHAR>()];
-    let res = GetFileInformationByHandleEx(
-        GetStdHandle(fd),
-        FileNameInfo,
-        &mut *name_info_bytes as *mut _ as *mut c_void,
-        name_info_bytes.len() as u32,
-    );
+    /// Mirrors winapi::um::fileapi::FILE_NAME_INFO, giving it a fixed length that
+    /// we can stack allocate
+    #[repr(C)]
+    struct FILE_NAME_INFO {
+        FileNameLength: DWORD,
+        FileName: [WCHAR; MAX_PATH]
+    }
```

Comment:
Do you know why the `winapi` definition isn't equivalent to this? I'm worried we're missing something given that `winapi` chose a different representation and it's not clear why to me. cc @retep998

Reply:
This is the way winapi normally models C's flexible array members, as it's typically the most flexible (heh) option. The actual max length of file name that `GetFileInformationByHandleEx` will write is controlled by the fourth parameter, which could well be less than the full `MAX_PATH` length (and could even be longer! It's possible to enable even longer path lengths, but our `pty`s shouldn't need to worry about that :))